src/factory-theia-client.ts

```diff
@@ -25,79 +25,74 @@ export enum ActionId {
  */
 export class FactoryTheiaClient {
 
-    private readonly cheFactoryApi: CheFactoryApi;
-
-    constructor() {
-        this.cheFactoryApi = new CheFactoryApi();
+    constructor(protected projectsRoot: string) {
     }
 
-    async onStart() {
+    async start() {
+        const factoryId = theia.env.getQueryParameter('factory-id');
 
-        const factory = await this.getFactoryFromCheMaster();
-        if (!factory) {
+        if (!factoryId || typeof factoryId !== 'string') {
+            return;
+        }
+
+        let factory: che.Factory;
+        try {
+            factory = await che.factory.getById(factoryId);
+        } catch (e) {
+            theia.window.showErrorMessage('Unable to get factory. ' + e.message);

```

Comment:
use of string literals ?

Reply:
Could you explain what do you mean?

Reply:
`` `Unable to get factory. ${e.message}` `` see https://basarat.gitbooks.io/typescript/docs/template-strings.html